app/views/shared/_message_notifier.html.haml

```diff
@@ -0,0 +1,33 @@
+%li
+  %ul.nav.pull-right
+    %li.dropdown.pull-right
+      = link_to "Notices (#{notifier.message_count})",
+        :notifications,
+        class: "dropdown-toggle",
+        data: { toggle: "dropdown" }
+
+      %ul.dropdown-menu
+        - if notifier.order_details_in_dispute?
```

Comment:
I had another thought: what if `notifier` responded to `each` and yields an object for each notification type that has notifications. It might mean that the notifier needs to know about `link_to` and the URL helpers, but it could clean up the view code quite a bit. It sort of turns the notifier object into more of a presenter.

```
%ul.dropdown-menu
  - notifier.each do |notifications|
    %li= notifications.link
```